### PR TITLE
[FixCode] Add a fixit to help users migrate to Swift 3 name convention.

### PR DESCRIPTION
**Pull request by nkcsgexi**
[FixCode] Add a fixit to help users migrate to Swift 3 name convention of enum cases. rdar://26887735

```
When users' referring to a enum case with a wrong name and we can find a correct enum case whose name differs from the wrong name only in capitalization, we replace the wrong name with the correct one.
```

#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

A smoke test on macOS does the following:
1. Builds the compiler incrementally.
2. Builds the standard library only for macOS. Simulator standard libraries and
   device standard libraries are not built.
3. lldb is not built.
4. The test and validation-test targets are run only for macOS. The optimized
   version of these tests are not run.

A smoke test on Linux does the following:
1. Builds the compiler incrementally.
2. Builds the standard library incrementally.
3. lldb is built incrementally.
4. The swift test and validation-test targets are run. The optimized version of these
   tests are not run.
5. lldb is tested.

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

…n of enum cases. rdar://26887735

When users' referring to a enum case with a wrong name and we can find a correct enum case whose name
differs from the wrong name only in capitalization, we replace the wrong name with the correct one.

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci test

**Comment by nkcsgexi** [comment]
@swift-ci smoke test OS X platform

**Comment by nkcsgexi** [comment]
@swift-ci smoke test OS X platform

**Comment by nkcsgexi** [comment]
@swift-ci smoke test OS X platform

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -72,6 +72,9 @@ ERROR(could_not_find_value_member,none,
 ERROR(could_not_find_type_member,none,
       "type %0 has no member %1", (Type, DeclName))
 
+ERROR(could_not_find_enum_case,none,
+      "enum type %0 has no case %1; do you mean %2", (Type, DeclName, DeclName))
```
Comment (CodaFi) [inline include/swift/AST/DiagnosticsSema.def]: Nitpick: This should be `did you mean`.

Comment (nkcsgexi) [inline include/swift/AST/DiagnosticsSema.def]: Good catch! I'm updating.

**Comment by nkcsgexi** [comment]
@swift-ci smoke test OS X platform

**Pull request merged**